elasticdl/python/ps/parameter_server.py

```diff
@@ -63,10 +68,16 @@ def prepare(self):
         self.logger.info("RPC Server started at port: %d", self.port)
 
     def run(self):
+        config.load_incluster_config()
+        api = client.CoreV1Api()
         try:
             while True:
-                # TODO: add loop break condition
                 time.sleep(30)
+                master_pod = api.read_namespaced_pod(
+                    namespace=self.namespace, name=self.master_name
+                )
+                if master_pod.status.phase == "Succeeded":
```

Comment:
If master pod status is `Failed`, ps also should stop.

Reply:
Done